### ISSUE TITLE
Discussion: the behavior of a new line with a backslash according to the position of the cursor

### ISSUE THREAD

**Issue by s417-lama** [fish-shell/fish-shell#4467]
# Environment

fish version: 2.6.0
OS: Ubuntu 16.04

# Description

As discussed in #613, currently the judgement whether a newline with a backslash is inserted is based on the position of the cursor.
However, when we edit commands after entering a backslash, a newline is not inserted and error message is shown.
For example,
```fish
$ hogeh(cursor is here)oge \
```
and press Enter key,
```fish
$ hogehoge \
fish: Unexpected end of string, incomplete escape sequence
hogehoge \
         ^
```
I think it's better to be able to insert a newline wherever the cursor is if the end of the line is a backslash. How do you think?

**Comment by ridiculousfish** [comment]
That sounds better than erroring at least :)

**Comment by s417-lama** [comment]
OK, I'll try to fix it.

**Comment by ridiculousfish** [comment]
Squash merged as 7004c30f954a760cbe8ac124ca7b3e2edeb0c57b. Thanks!!
